hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CSQueue.java

```diff
@@ -151,6 +151,12 @@
    * @return current run-state
    */
   public QueueState getState();
+
+  /**
+   * Get the max-parallel-applications property of the queue
+   * @return max-parallel-applications
+   */
+  public int getMaxParallelApps();
```

Comment:
I am not familiar with the convention whether we are allowed to add new methods on CSQueue, as it is marked as Stable. Check weight, which has been added recently, but was not introduced in CSQueue either.

Reply:
CapacitySchedulerQueueInfo(CapacityScheduler cs, CSQueue q) {
    weight = q.getQueueCapacities().getWeight();     // the getWeight method was already implemented
    maxParallelApps = q.getMaxParallelApps();          // the CSQueue interface must be extended to implement this feature

Reply:
I think it's fine to add getters because it only extends the interface but won't break backwards compatibility.

Looking at the javadoc of InterfaceStability.Stable: 
```
  /**
   * Can evolve while retaining compatibility for minor release boundaries.; 
   * can break compatibility only at major release (ie. at m.0).
   */
  @Documented
  @Retention(RetentionPolicy.RUNTIME)
  public @interface Stable {};
```
This also suggests the same.

Reply:
Thanks for the feedback, resolving this thread.